python311Packages.rlax: fix build

GaetanLepage opened the pull request:
## Description of changes

An attempt at fixing `plotnine` to have `bsuite` and thus `rlax` working.

cc @onny 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python311Packages.plotnine: 0.12.3 -> 0.12.4

Changelog: https://github.com/has2k1/plotnine/releases/tag/v0.12.4

GaetanLepage committed:
python311Packages.rlax: fix build

samuela commented on pkgs/development/python-modules/mizani/default.nix:
```diff
@@ -12,16 +12,16 @@
   pname = "mizani";
-  version = "0.10.0";
-  format = "pyproject";
+  version = "0.9.3";
```
why is a downgrade necessary here?

GaetanLepage replied:
In the end it was not.
I was testing things around.

GaetanLepage replied:
Actually, plotnine==0.12.4 is not compatible with mizani==0.10.0.
As this downgrade does not break anything (but fixes plotnine), I guess that it is justified.

GaetanLepage replied:
Also, only two packages depend on `mizani`: `bsuite` and `plotnine` which latest versions both work fine with `mizani`==0.9.3

NickCao replied:
The downgrade is reasonable, but the commit message is backwards.

GaetanLepage replied:
Indeed, fixed.

samuela commented on pkgs/development/python-modules/plotnine/default.nix:
```diff
@@ -1,7 +1,8 @@
 { lib
-, adjusttext
 , buildPythonPackage
 , fetchPypi
```
do we still need `fetchPypi`?

GaetanLepage replied:
No, removed.

samuela commented on pkgs/development/python-modules/plotnine/default.nix:
```diff
@@ -46,14 +50,15 @@ buildPythonPackage rec {
 
   preCheck = ''
     export HOME=$(mktemp -d)
+
+    pytest -s tests/test_geom_violin.py
+    exit 1
```
looks like this needs to be removed?

GaetanLepage replied:
Yes, this was for debugging.

samuela commented on pkgs/development/python-modules/plotnine/default.nix:
```diff
@@ -75,6 +80,9 @@ buildPythonPackage rec {
     "tests/test_coords.py"
     "tests/test_annotation_logticks.py"
+
+    # Linting / formatting: useless as it has nothing to do with the package functionning
```
does this test fail for some reason? let's default to running more tests rather than fewer, even if they do not seem directly relevant

NickCao replied:
I guess they just unnecessarily pull in the linters.

GaetanLepage replied:
Yes, I clarified the comment.

samuela commented on pkgs/development/python-modules/rlax/default.nix:
```diff
@@ -1,39 +1,69 @@
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-C3nFOv/zxvAoz6WZ0RAZffzEbxIx/XrGabO4QPxrik8=";
+  # src = fetchPypi {
+  #   inherit pname version;
+  #   hash = "sha256-C3nFOv/zxvAoz6WZ0RAZffzEbxIx/XrGabO4QPxrik8=";
+  # };
```
```suggestion
```

samuela commented on pkgs/development/python-modules/rlax/default.nix:
```diff
@@ -1,39 +1,69 @@
 
-  buildInputs = [
+  patches = [
+    (fetchpatch {
+      name = "replace-deprecated-chex-assertions";
+      url = "https://github.com/google-deepmind/rlax/commit/30e7913a1102667137654d6e652a6c4b9e9ba1f4.patch";
+      hash = "sha256-OPnuTKEtwZ28hzR1660v3DcktxTYjhR1xYvFbQvOhgs=";
+    })
```
Let's add a comment explaining why this patch is necessary and when it can be safely removed

GaetanLepage replied:
Done !

samuela commented on pkgs/development/python-modules/rlax/default.nix:
```diff
@@ -1,39 +1,69 @@
     tensorflow-probability
   ];
 
+  # buildInputs = [
+  #   chex
+  #   jaxlib
+  #   distrax
+  #   tensorflow-probability
+  # ];
```
```suggestion
```

GaetanLepage replied:
Removed.

samuela commented on pkgs/development/python-modules/rlax/default.nix:
```diff
@@ -1,39 +1,69 @@
   nativeCheckInputs = [
-    bsuite
-    dm-env
+    # bsuite
+    # dm-env
```
```suggestion
```

samuela commented on pkgs/development/python-modules/rlax/default.nix:
```diff
@@ -1,39 +1,69 @@
+    # dm-env
     dm-haiku
-    frozendict
+    # frozendict
```
```suggestion
```

samuela reviewed (dismissed):
looks like this is still a WIP, but my review was requested so providing some feedback in its current state

imincik commented:
@GaetanLepage , great thanks for this PR. I see plotnine, bsuite and r-lax failing after all geospatial packages updates. I agree with all points raised by @samuela .

GaetanLepage commented:
> @GaetanLepage , great thanks for this PR. I see plotnine, bsuite and r-lax failing after all geospatial packages updates. I agree with all points raised by @samuela .

Yes, I think that `bsuite` and `rlax` are only broken because of theuir dependency `plotnine` being itself broken.
I tried to have a look at the latter. Most of the tests pass (so no hard crash or similar error), but I was not able to understand why those tests were failing.
They are not failing when I run them on a non-Nix environment (using a venv).
Most of them consist in comparing generated images of plots to expected ones. Not really easy to debug as I can't see the generated results in the build environment.

imincik commented:
> Not really easy to debug as I can't see the generated results in the build environment.
You can try to interactively run the build using [nix-develop-interactive.bash](https://github.com/imincik/nix-utils/blob/master/nix-develop-interactive.bash). Using this script you can manually run and re-run phases and you have full access to all files.

USAGE: 
```
mkdir dev; cd dev
nix develop nixpkgs#<PACKAGE>
source nix-develop-interactive.bash
```

GaetanLepage commented:
> You can try to interactively run the build using [nix-develop-interactive.bash](https://github.com/imincik/nix-utils/blob/master/nix-develop-interactive.bash).

Thanks ! I wonder how I was not already aware of this tool.
So I was able to check.
The images are indeed different...by a minuscule amount. If you quickly cycle between expected and actual images, you see that a few elements are 1 pixel shifted.
Now, to know why this is happening... well I have no idea. Maybe some dependency being too old...

GaetanLepage commented:
I have disabled those tests.

GaetanLepage commented:
I have disabled those tests.

GaetanLepage committed:
python311Packages.mizani: 0.10.0 -> 0.9.3

Downgrade as latest version plotnine (0.12.4) is not compatible with mizani==0.10.0

GaetanLepage commented:
Result of `nixpkgs-review pr 275438` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.mizani</li>
    <li>python310Packages.mizani.dist</li>
    <li>python310Packages.plotnine</li>
    <li>python310Packages.plotnine.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.mizani</li>
    <li>python311Packages.mizani.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
  </ul>
</details>

NickCao reviewed (approved):
diff LGTM, nixpkgs-review happy. Waiting for ofborg.

GaetanLepage commented:
Result of `nixpkgs-review pr 275438` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python310Packages.bsuite</li>
    <li>python310Packages.bsuite.dist</li>
    <li>python310Packages.mizani</li>
    <li>python310Packages.mizani.dist</li>
    <li>python310Packages.plotnine</li>
    <li>python310Packages.plotnine.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.mizani</li>
    <li>python311Packages.mizani.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
  </ul>
</details>

imincik reviewed (approved):
Great work, thanks !
